apps/files/models.py

```diff
@@ -378,24 +380,6 @@ def get_localepicker(self):
         statsd.timing('files.extract.localepicker', (end * 1000))
         return res
 
-    @property
-    def is_signed(self):
-        """Return True if the file has been signed.
-
-        This will simply check the signature filenames, and assume that if
-        they're named "mozilla.*" then the xpi has been signed by us.
-
-        This is in no way a perfect or correct solution, it's just the way we
-        do it until we decide to inspect/walk the certificates chain to
-        validate it comes from Mozilla."""
-        try:
-            with zipfile.ZipFile(self.file_path, mode='r') as zf:
-                filenames = set(zf.namelist())
-        except (zipfile.BadZipfile, IOError):
-            filenames = set()
-        return set(['META-INF/mozilla.rsa', 'META-INF/mozilla.sf',
-                    'META-INF/manifest.mf']).issubset(filenames)
```

Comment:
Just an idea, no need to use it but curious if you used a null boolean field, null could be you haven't checked the file yet, and true/false mean you know the value. So if the db field isn't true/false fall back to doing the above -- opening the zip, check it, etc. then set the value for the next time?